spec/date_parsing_spec.rb

```diff
@@ -0,0 +1,517 @@
+    'view of approximately 1848, published about 1865' => ['1848', '1865']
+  }
+  # example string as key, expected parsed value as value
+  multiple_years_4_digits_once = {
+    '1918-20' => ['1918', '1919', '1920'],   # vs. 1961-04
+    '1965-8' => ['1965', '1966', '1967', '1968'], # revs
+    '[1846-51]' => ['1846', '1847', '1848', '1849', '1850', '1851']
+  }
+  # example string as key, expected parsed value as value
+  decade_only_4_digits = {
+    'early 1890s' => ['1890', '1891', '1892', '1893', '1894', '1895', '1896', '1897', '1898', '1899'],
+    '1950s' => ['1950', '1951', '1952', '1953', '1954', '1955', '1956', '1957', '1958', '1959'],
+    "1950's" => ['1950', '1951', '1952', '1953', '1954', '1955', '1956', '1957', '1958', '1959']
+  }
+  decade_only = {
+    '156u' => ['1560', '1561', '1562', '1563', '1564', '1565', '1566', '1567', '1568', '1569'],
+    '167-?]' => ['1670', '1671', '1672', '1673', '1674', '1675', '1676', '1677', '1678', '1679'],
+    '[171-?]' => ['1710', '1711', '1712', '1713', '1714', '1715', '1716', '1717', '1718', '1719'],
+    '[189-]' => ['1890', '1891', '1892', '1893', '1894', '1895', '1896', '1897', '1898', '1899'],
+    'ca.170-?]' => ['1700', '1701', '1702', '1703', '1704', '1705', '1706', '1707', '1708', '1709'],
+    '200-?]' => ['2000', '2001', '2002', '2003', '2004', '2005', '2006', '2007', '2008', '2009'],
+    '186?' => ['1860', '1861', '1862', '1863', '1864', '1865', '1866', '1867', '1868', '1869'],
+    '195x' => ['1950', '1951', '1952', '1953', '1954', '1955', '1956', '1957', '1958', '1959']
+  }
+  brackets_in_middle_of_year = {
+    '169[5]' => '1695',
+    'October 3, [18]91' => '1891'
+  }
+
+  context '*sortable_year_from_date_str' do
+    single_year
+      .merge(specific_month)
+      .merge(specific_day)
+      .merge(invalid_but_can_get_year).each do |example, expected|
+      it "gets #{expected} from #{example}" do
+        expect(Stanford::Mods::DateParsing.sortable_year_from_date_str(example)).to eq expected
+      end
+    end
+
+    multiple_years.each do |example, expected|
+      it "gets #{expected.first} from #{example}" do
+        expect(Stanford::Mods::DateParsing.sortable_year_from_date_str(example)).to eq expected.first
+      end
+    end
+
+    specific_day_ruby_parse_fail.each do |example, expected|
+      it "gets #{expected} from #{example}" do
+        expect(Stanford::Mods::DateParsing.sortable_year_from_date_str(example)).to eq expected
+      end
+    end
+
+    multiple_years_4_digits_once
+      .merge(decade_only_4_digits).each do |example, expected|
+      it "gets #{expected} from #{example}" do
+        expect(Stanford::Mods::DateParsing.sortable_year_from_date_str(example)).to eq expected.first
+      end
+    end
+
+    unparseable
+      .push(*brackets_in_middle_of_year.keys)
+      .push(*specific_day_2_digit_year.keys)
+      .push(*decade_only.keys)
+      .push(*century_only.keys).each do |example|
+      it "nil for #{example}" do
```

Comment:
hrmm.  My idea is that the DateParsing methods are intended for specific types of date parsing, and there will be an uber method (or a few of them) in origin_info that will call the DateParsing methods.   

It may be that there is a better way to indicate "failure is ok/expected for these categories of dates" since my intent is to indicate that the given method works for some things and not for others.   

And the uber method will want to show that it properly handles all the cases -- and I'm not sure I can refer to the data constants in the date parsing spec  from  an origin info  spec ...